sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -0,0 +1,180 @@
+  blobClient: CopyPollerBlobClient;
+  copyId?: string;
+  copyProgress?: string;
+  copySource: string;
+}
+
+/**
+ * The PollOperation responsible for:
+ *  - performing the initial startCopyFromURL
+ *  - checking the copy status via getProperties
+ *  - cancellation via abortCopyFromURL
+ */
+export interface BlobBeginCopyFromURLPollOperation
+  extends PollOperation<BlobBeginCopyFromUrlPollState, BlobStartCopyFromURLResponse> {}
+
+/**
+ *
+ */
+export interface BlobBeginCopyFromUrlPollerOptions {
+  blobClient: CopyPollerBlobClient;
+  copySource: string;
+  intervalInMs?: number;
+  onProgress?: (state: BlobBeginCopyFromUrlPollState) => void;
+  resumeFrom?: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ *
+ */
+export class BlobBeginCopyFromUrlPoller extends Poller<
+  BlobBeginCopyFromUrlPollState,
+  BlobStartCopyFromURLResponse
+> {
+  public intervalInMs: number;
+
+  constructor(options: BlobBeginCopyFromUrlPollerOptions) {
+    const { blobClient, copySource, intervalInMs = 15000, onProgress, resumeFrom } = options;
+
+    let state: BlobBeginCopyFromUrlPollState | undefined;
+
+    if (resumeFrom) {
+      state = JSON.parse(resumeFrom).state;
+    }
+
+    const operation = makeBlobBeginCopyFromURLPollOperation({
+      ...state,
+      blobClient,
+      copySource
+    });
+
+    super(operation);
+
+    if (typeof onProgress === "function") {
+      this.onProgress(onProgress);
+    }
+
+    this.intervalInMs = intervalInMs;
+  }
+
+  async delay(): Promise<void> {
+    return delay(this.intervalInMs);
+  }
+}
+
+const cancel: BlobBeginCopyFromURLPollOperation["cancel"] = async function cancel(
+  this: BlobBeginCopyFromURLPollOperation,
+  options = {}
+) {
+  const state = this.state;
+  const { copyId } = state;
+  if (state.completed) {
+    return makeBlobBeginCopyFromURLPollOperation(state);
+  }
+
+  if (!copyId) {
+    state.cancelled = true;
+    return makeBlobBeginCopyFromURLPollOperation(state);
+  }
+
+  try {
+    // TODO: Support abort options
+    await state.blobClient.abortCopyFromURL(copyId, {
+      abortSignal: options.abortSignal
+    });
+    state.cancelled = true;
+  } catch (err) {
+    // Not sure what to do in this case... treat as not cancelled?
+  }
+
+  return makeBlobBeginCopyFromURLPollOperation(state);
+};
+
+const update: BlobBeginCopyFromURLPollOperation["update"] = async function update(
+  this: BlobBeginCopyFromURLPollOperation,
+  options = {}
+): Promise<BlobBeginCopyFromURLPollOperation> {
+  const state = this.state;
+  const { blobClient, copySource } = state;
+
+  // TODO: Logic when state isn't started
+  if (!state.started) {
+    state.started = true;
+    const result = await blobClient.startCopyFromURL(copySource);
+
+    // copyId is needed to abort
+    state.copyId = result.copyId;
+    if (result.copyStatus === "success") {
+      state.result = result;
+      state.completed = true;
+    }
+    // TODO: Check other statuses?
```

Comment:
It's probably worth checking the failed status (I'm not sure when it will report a failure vs throw an error), otherwise that's handled on on the next call to `poll` in the else if.